paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -125,61 +122,61 @@ internal abstract class BaseSheetViewModel(
+    internal val headerText = MutableStateFlow<String?>(null)
 
     /**
      * Request to retrieve the value from the repository happens when initialize any fragment
      * and any fragment will re-update when the result comes back.
      * Represents what the user last selects (add or buy) on the
      * [PaymentOptionsActivity]/[PaymentSheetActivity], and saved/restored from the preferences.
      */
-    private val _savedSelection =
-        savedStateHandle.getLiveData<SavedSelection>(SAVE_SAVED_SELECTION)
-    private val savedSelection: LiveData<SavedSelection> = _savedSelection
+    private val savedSelection: StateFlow<SavedSelection> =
+        savedStateHandle.getStateFlow<SavedSelection>(SAVE_SAVED_SELECTION, SavedSelection.None)
 
-    private val _transition = MutableLiveData<Event<TransitionTarget>?>(null)
-    internal val transition: LiveData<Event<TransitionTarget>?> = _transition
+    private val _transition = MutableStateFlow<Event<TransitionTarget>?>(null)
+    internal val transition: StateFlow<Event<TransitionTarget>?> = _transition
 
-    private val _liveMode = savedStateHandle.getLiveData<Boolean>(SAVE_STATE_LIVE_MODE)
-    internal val liveMode: LiveData<Boolean> = _liveMode
+    private val _liveMode = MutableStateFlow(savedStateHandle[SAVE_STATE_LIVE_MODE] ?: false)
+    internal val liveMode: StateFlow<Boolean> = _liveMode
 
-    private val _selection = savedStateHandle.getLiveData<PaymentSelection>(SAVE_SELECTION)
-    internal val selection: LiveData<PaymentSelection?> = _selection
+    internal val selection: StateFlow<PaymentSelection?> =
+        savedStateHandle.getStateFlow(SAVE_SELECTION, null)
 
-    private val editing = MutableLiveData(false)
+    private val editing = MutableStateFlow(false)
 
-    val processing: LiveData<Boolean> = savedStateHandle.getLiveData<Boolean>(SAVE_PROCESSING)
+    val processing: StateFlow<Boolean> = savedStateHandle
+        .getStateFlow(SAVE_PROCESSING, false)
 
-    private val _contentVisible = MutableLiveData(true)
-    internal val contentVisible: LiveData<Boolean> = _contentVisible.distinctUntilChanged()
+    private val _contentVisible = MutableStateFlow(true)
+    internal val contentVisible: StateFlow<Boolean> = _contentVisible
 
     /**
      * Use this to override the current UI state of the primary button. The UI state is reset every
      * time the payment selection is changed.
      */
-    private val _primaryButtonUIState = MutableLiveData<PrimaryButton.UIState?>()
-    val primaryButtonUIState: LiveData<PrimaryButton.UIState?> = _primaryButtonUIState
+    private val _primaryButtonUIState = MutableStateFlow<PrimaryButton.UIState?>(null)
+    val primaryButtonUIState: StateFlow<PrimaryButton.UIState?> = _primaryButtonUIState
 
-    private val _primaryButtonState = MutableLiveData<PrimaryButton.State>()
-    val primaryButtonState: LiveData<PrimaryButton.State> = _primaryButtonState
+    private val _resetPrimaryButtonUI = MutableStateFlow(0L)
```

Comment:
Is there a better way to force a state flow to update? I am using a long and updating the value each time it needs to be updated.

Reply:
Looks like the state is being held in the Activities. Rather than here in the view model, so this is essentially just being used as a notifier. If the state was being updated here, it would be clear what was happening, and this wouldn't be needed.

That being said, I don't think it's appropriate to make that change in this PR. But seems like a good thing to do as a follow up.